src/main/java/com/glodblock/github/proxy/CommonProxy.java

```diff
@@ -1,42 +1,129 @@
 import cpw.mods.fml.common.event.FMLPreInitializationEvent;
+import cpw.mods.fml.common.eventhandler.SubscribeEvent;
+import cpw.mods.fml.common.gameevent.TickEvent;
 
 public class CommonProxy {
 
     public final FCNetworkWrapper netHandler = new FCNetworkWrapper(FluidCraft.MODID);
 
+    public CommonProxy() {
+        MinecraftForge.EVENT_BUS.register(this);
+        FMLCommonHandler.instance().bus().register(this);
+    }
+
+    @SuppressWarnings("unchecked")
+    @SubscribeEvent
+    public void pickupEvent(EntityItemPickupEvent e) {
+        try {
+            if (Platform.isClient() || e.entityPlayer == null) return;
+            EntityPlayer player = e.entityPlayer;
+            EntityItem itemEntity = e.item;
+            ItemStack stack = itemEntity.getEntityItem();
+            World world = player.getEntityWorld();
+
+            ImmutablePair<Integer, ItemStack> result = Util.Wireless.getUltraWirelessTerm(player);
+            if (result == null) return;
+            final ItemStack wirelessTerm = result.getRight();
+            ItemMagnetCard.Mode mode = ItemMagnetCard.getMode(wirelessTerm);
+            if (!Util.Wireless.hasMagnetCard(wirelessTerm) || mode != ItemMagnetCard.Mode.ME) return;
+            IGridNode gridNode = Util.Wireless.getWirelessGrid(wirelessTerm);
+            if (gridNode == null || !Util.Wireless.rangeCheck(wirelessTerm, player, gridNode)) return;
+            WirelessMagnetCardFilterInventory inv = new WirelessMagnetCardFilterInventory(
+                    wirelessTerm,
+                    result.getLeft(),
+                    gridNode,
+                    player);
+
+            IItemList<IAEItemStack> filteredList = inv.getAEFilteredItems();
+            IAEItemStack ais = AEApi.instance().storage().createItemStack(stack);
+            if (inv.getListMode() == ItemMagnetCard.ListMode.WhiteList) { // whitelisting
+                if (!filteredList.isEmpty() && inv.isItemFiltered(stack, filteredList)) {
+                    if (inv.doInject(ais, itemEntity, world)) {
+                        stack = null;
+                        itemEntity.setDead();
+                        e.setCanceled(true);
+                    }
+                }
+            } else if (inv.getListMode() == ItemMagnetCard.ListMode.BlackList) {
+                if (!inv.isItemFiltered(stack, filteredList)) {
+                    if (inv.doInject(ais, itemEntity, world)) {
+                        stack = null;
+                        itemEntity.setDead();
+                        e.setCanceled(true);
+                    }
+                }
+            }
+
+        } catch (NullPointerException ex) {
+            ex.printStackTrace();
+        }
+    }
+
+    @SubscribeEvent
+    public void tickEvent(TickEvent.PlayerTickEvent e) {
+        EntityPlayer player = e.player;
+        ImmutablePair<Integer, ItemStack> result = Util.Wireless.getUltraWirelessTerm(player);
```

Comment:
Check that all logic work fine when terminal placed in baubles slot, Include mode switch